### DISCUSSION TITLE
Learning VSC for IBMi - debugging issue

### DISCUSSION THREAD

**Discussion by dwilkins-curbstone** [codefori/vscode-ibmi#1607] [Q&A]
I followed the IBMi debug install and walkthrough but when I try to start a debug session on a program I get the following message and after some research I found on the merlin page that it could mean that the debugging service isn't started. But, I thought I started it on when I went through the setup and the walkthrough page. 

![Capture](https://github.com/codefori/vscode-ibmi/assets/100879024/39001a8a-38b8-47ca-886e-8ede04fed29b)

How do I start the debugging service using VSC? I'm thinking this isn't the same as running command STRDBGSRV on the IBMi command line. But I'm not 100% sure of that. Any help is appreciated. The IBMi is listening on port 8005.

**Comment by worksofliam** [comment]
Hi @dwilkins-curbstone 

Check out our docs on the Debugger. You can see how to start it up here, but the Walkthrough will help you start it up as I am sure you are aware.

https://codefori.github.io/docs/#/pages/developing/debug/debug

**Reply by dwilkins-curbstone** [reply]
Thank you Liam. The service was actually started but I told it to end and restart. I got the same error when trying to start a debug session. I also received this error that I didn't notice the last time. 
![image](https://github.com/codefori/vscode-ibmi/assets/100879024/a323d497-f3e7-4bae-a885-06ff2e5ad44f)

So, it doesn't seem the first error was because it wasn't started since it asked if I wanted to restart the service. Does this new message give any clues as to what I need to do to make this work?

**Reply by dwilkins-curbstone** [reply]
I connect to this system via a VPN so I'm wondering if that has something to do with the debug session not working. The service is running I just keep getting the same errors or a timeout.

**Comment by dwilkins-curbstone** [comment]
Does anyone have any tips or advice for me as to what I need to look for on our iseries that could be causing the issue?

**Reply by mahesh-patel-jangili** [reply]
ctrl + shift p and start debug session from command pallet in  vscode.
![image](https://github.com/codefori/vscode-ibmi/assets/130437537/043975bd-4b16-47ad-aafb-dfba2e3f84fa)

i am also using vpn to connect my ibm i so its not a problem with vpn

**Reply by NorskieRider** [reply]
On the IBM i after you run STRDBGSVR you should see that port 8005 is open (use netstat opt 3 to validate that). If that port isn't open then check job QDB5ROUTER for error messages.
